### PR TITLE
Setup node_modules caching in CI

### PR DESCRIPTION
**Pull request by Peterabsolon**
Closes https://github.com/nearform/brokeneck/issues/48

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,12 +15,30 @@ jobs:
...
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: |
+            **/node_modules
```
Comment (simoneb) [inline .github/workflows/ci.yml:34]: can you make sure that this matches __both__ the root node_modules and that of the nested packages? In the docs of the cache action for lerna, it shows them as two different patterns, maybe for a reason

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,12 +15,30 @@ jobs:
...
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      
       - run: npx lerna bootstrap
       - run: npx lerna run lint
       - run: npx lerna run build
       - run: npx lerna run test
```
Comment (simoneb) [inline .github/workflows/ci.yml:43]: even if not really related to caching, I think running npx is downloading lerna each single time. it might be worth switching to a way of running it that doesn't involve using npx

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -15,12 +15,31 @@ jobs:
...
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-
+      
+      - run: yarn install --frozen-lockfile
```
Comment (simoneb) [inline .github/workflows/ci.yml:41]: let's keep it consistent with the following steps and run lerna bootstrap. with the change you've done in lerna.json it should already provide the frozen lockfile option. also, I'm not sure that this is doing what lerna bootstrap does and even though it may be doing it, it's not obvious if it is. lerna boostrap is doing more than what yarn install does, even when using workspaces. lerna is symlinking packages, and that's the way we should run it.

in other words, lerna boostrap implies yarn install, but not the other way around, or at least it's not obvious whether it is doing that or not.

**Review by simoneb — COMMENTED** [review]
Only one comment, otherwise it looks good to me.

File: packages/brokeneck-react/package.json [hunk 4]
```diff
@@ -29,6 +29,7 @@
     "graphql-hooks": "^5.0.0",
     "lodash.debounce": "^4.0.8",
     "lodash.isnil": "^4.0.0",
+    "lodash.set": "^4.3.2",
```
Comment (simoneb) [inline packages/brokeneck-react/package.json:32]: do we need this?

Comment (Peterabsolon) [reply packages/brokeneck-react/package.json:32]: we don't, good catch

**Review by simoneb — APPROVED** [review]
